packages/cli/src/utils.js

```diff
@@ -18,6 +18,16 @@ const _require = esm(module, {
 const getRootDir = argv => path.resolve(argv._[0] || '.')
 const getNuxtConfigFile = argv => path.resolve(getRootDir(argv), argv['config-file'])
 
+export function getLockPath(dir) {
+  return path.resolve(dir || process.cwd())
```

Comment:
Current nuxt repositories are not configured to gitignore such lock file. So we should either use `node_modules` (rootDir/node_modules) or `.nuxt` (buildDir). Which i suggest the second one as there may be multi projects with the same rootDir.

Reply:
My original plan was to use `buildDir` indeed, but I didnt want to wait with locking until nuxt was initialised. And `buildDir` / `.nuxt` could not exist yet and didnt want to go through the trouble of having to mkdirp that folder.
In a normal project the lock dir is on the same level as the project dir with the current implementation, so if you have `/var/www/projects/my-huxt-project/` which is a git repo then the lock folder will be `/var/www/projects/my-huxt-project.lock/` (which is outside the git repo).

Reply:
@pimlie That would be something worst (creating something outside of project). What if we move lock creating logic intro `core/Nuxt` class? This way programmatic usage would benefit from this feature and also we have a chance to easily use `this.options` to create buildDir.

Reply:
Hmm, moving to `core/Nuxt` sounds a bit like you suggest you could only ever have one lock within your nuxt project. I'd like to keep the idea open for now that you could use multiple locks.

We could move locking methods to `common`, add a `options.build.lock` property and implement it by default for the builder on `buildDir`?

Reply:
Another solution would be to (optionally) factor the `getNuxtConfig` out of the `core/Nuxt` constructor. Especially now since the config is a separate package anyway, we could implement similar logic as with NuxtCommand, so a `class NuxtConfig` with a static `NuxtConfig.from(options)`. That way we could check in core/Nuxt whether defaults have been applied or not and in cli you dont have to instantiate core/Nuxt just to find the buildDir.